File: 2022/Day3/Puzzel2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

int main()
{
    std::ifstream inputFile("input.txt");
    std::vector<std::string> input = {};
    std::string temp;
    int priority = 0;
    while( std::getline(inputFile, temp))
    {
        input.push_back(temp);
    }
    for(int lineIndex = 0; lineIndex<input.size(); lineIndex+=3)
    {
        bool found = false;
        for( int x=0; x<input[lineIndex].size()&&!found; x++){
            char elf1 = input[lineIndex][x];
            for( int y=0; y<input[lineIndex+1].size()&&!found; y++){
                char elf2 = input[lineIndex+1][y];
                if (elf1 == elf2)
                {
                    for( auto elf3: input[lineIndex+2])
                    {
                        if (elf2 == elf3)
                        {
                            if (elf3 >= 'a' && elf3 <= 'z')
                            {
                                priority+=elf3-'a'+1;
                                found = true;
                                break;
                            }
                            else if(elf3 >= 'A' && elf3 <= 'Z')
                            {
                                priority+=elf3-'A'+26+1;
                                found = true;
                                break;
                            }
                        }
                    }
                }
            }
        }
    }
    std::cout << "Priority = " << priority << '\n';
}

File: 2022/Day03/Puzzel2.py
priority = 0;
input = []

with open("input.txt") as inputFile:
    for line in inputFile:
        input.append(line.strip())


for i in range(0, len(input), 3):
    elf1 = input[i]
    elf2 = input[i+1]
    elf3 = input[i+2]
    found = False;
    for itemElf1 in elf1:
        if found:
            break
        for itemElf2 in elf2:
            if found:
                break
            if itemElf1 == itemElf2 :
                for itemElf3 in elf3:
                    if found:
                        break
                    if itemElf3 == itemElf1:
                        found = True
                        if itemElf3 >= 'a' and itemElf3 <= 'z':
                            priority += ord(itemElf3)-ord('a')+1        
                        elif itemElf3 >= 'A' and itemElf3 <= 'Z':
                            priority += ord(itemElf3)-ord('A')+26+1            
                        else:
                            print( f"unexpected character found: {itemElf3}\n")

print( f"Priority = {priority}" )
